File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        sort(nums.begin(),nums.end());
        int ans=1,tmp=1;
        for(int i=1;i<nums.size();i++){
            if(nums[i]-nums[i-1]==1)tmp++;
            else if(nums[i]-nums[i-1]>1)tmp=1;  
            ans=max(ans,tmp);
        }
        if(nums.size()==0)return 0;
        return ans;
    }
};

File: Data Structures & Algorithms/longest-increasing-subsequence/submission-3.py
from bisect import bisect_left
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        #tails
        #make a "fake" increasing subsequence which keeps increeasing and if we find something small, we replace it
        n = len(nums)
        tails=[]
        ans=0
        for i in range(n):
            ind = bisect_left(tails,nums[i])
            if ind == len(tails):
                tails.append(nums[i])
            else:
                tails[ind]=nums[i]
        return len(tails)